### DISCUSSION TITLE
How to depend on modified sentry-android source code?

### DISCUSSION THREAD

**Discussion by fzyzcjy** [getsentry/sentry-java#2539] [Q&A]
### Description

Hi thanks for the Sentry! I want to modify sentry android code a little bit and then use it in my project. The question is how can I do that? As sentry-java even depends on sentry-native, I guess there should be some automation that can be used to make the usage easy, instead of manually compiling everything and push them to maven etc.

**Comment by romtsn** [comment]
I guess it depends on your modifications. If they do not involve NDK/sentry-native changes, you could simply keep them, and only replace the affected modules using Gradle dependency substitution. Assuming your changes are in `sentry-android-core`, you could do:

```gradle
configurations.all {
    resolutionStrategy.dependencySubstitution {
        substitute module("io.sentry:sentry-android-core") using project(":custom-sentry-android-core") because "we have a critical bugfix and Sentry has not released it yet"
    }
}
```
